### PR TITLE
Include hooks listlastFire api support

### PR DESCRIPTION
**Pull request by Biboswan**

### CODE REVIEW

**Comment by Biboswan** [comment]
I have checked query's response in the graghql playground

File: services/web-server/src/graphql/Hooks.graphql [hunk 1]
```diff
@@ -33,6 +33,12 @@ enum HookFireResult {
   NO_FIRE
 }
 
+enum HookFiredBy {
+  SCHEDULE
+  TRIGGERHOOK
```
Comment (helfi92) [inline services/web-server/src/graphql/Hooks.graphql]: Rename to `TRIGGER_HOOK`.

File: services/web-server/src/graphql/Hooks.graphql [hunk 2]
```diff
@@ -33,6 +33,12 @@ enum HookFireResult {
...
+  TRIGGERHOOKWITHTOKEN
```
Comment (helfi92) [inline services/web-server/src/graphql/Hooks.graphql]: Rename to `TRIGGER_HOOK_WITH_TOKEN`.

File: services/web-server/src/graphql/Hooks.graphql [hunk 3]
```diff
@@ -89,11 +95,22 @@ input HookInput {
   triggerSchema: JSON
 }
 
+type HookLastFire {
+  hookGroupId: ID!,
+  hookId: ID!,
+  firedBy: HookFiredBy!,
```
Comment (helfi92) [inline services/web-server/src/graphql/Hooks.graphql]: Should `firedBy` be required? I don't see it set as required in https://github.com/taskcluster/taskcluster/blob/30372ec731730592a7bf22d4b349a61f75663883/services/hooks/schemas/v1/list-lastFires-response.yml#L41-L47.

If it's not required, then it should be `firedBy: HookFiredBy,`. If it's required, then `list-lastFires-response.yml
` needs to be updated.

File: services/web-server/src/graphql/Hooks.graphql [hunk 4]
```diff
@@ -89,11 +95,22 @@ input HookInput {
...
+  result:HookFireResult!,
+  error: String!,
+}
+
 extend type Query {
   hookGroups(filter: JSON): [HookGroup]
   hooks(hookGroupId: ID!, filter: JSON): [Hook]
   hook(hookGroupId: ID!, hookId: ID!): Hook
   hookStatus(hookGroupId: ID!, hookId: ID!): HookStatus
+  hookLastFires(hookGroupId: ID!, hookId: ID!): [HookLastFire]
```
Comment (helfi92) [inline services/web-server/src/graphql/Hooks.graphql]: When the response is an array, we should allow a user to filter on the result. Let's add an additional optional parameter `filter: JSON`. See the `diff` for how to fully integrate the filter feature.

Comment (Biboswan) [reply services/web-server/src/graphql/Hooks.graphql]: So I think of adding a searchbar for finding apt lastfires by success, time etc.

Comment (helfi92) [reply services/web-server/src/graphql/Hooks.graphql]: The graphQL endpoint should still have a filter parameter. It doesn't mean we will use it but as a convention, for all response returning an array, we should expose a filter parameter.

File: services/web-server/src/graphql/Hooks.graphql [hunk 5]
```diff
@@ -89,11 +95,22 @@ input HookInput {
   triggerSchema: JSON
 }
 
+type HookLastFire {
+  hookGroupId: ID!,
+  hookId: ID!,
+  firedBy: HookFiredBy!,
+  taskId: String!,
+  taskCreateTime: DateTime!,
+  result:HookFireResult!,
```
Comment (helfi92) [inline services/web-server/src/graphql/Hooks.graphql]: Add a space after the column:

`result: HookFireResult!,`

File: services/web-server/src/resolvers/Hooks.js [hunk 6]
```diff
@@ -40,6 +45,9 @@ export default {
     hookStatus(parent, { hookGroupId, hookId }, { loaders }) {
       return loaders.hookStatus.load({ hookGroupId, hookId });
     },
+    hookLastFires(parent, {hookGroupId, hookId, filter}, {loaders}) {
```
Comment (helfi92) [inline services/web-server/src/resolvers/Hooks.js]: Add a space after `{` and before `}` when destructuring:

```js
hookLastFires(parent, { hookGroupId, hookId, filter }, { loaders }) {
```

File: services/web-server/src/resolvers/Hooks.js [hunk 7]
```diff
@@ -40,6 +45,9 @@ export default {
...
+      return loaders.hookLastFires.load({hookGroupId, hookId, filter});
```
Comment (helfi92) [inline services/web-server/src/resolvers/Hooks.js]: Add a space after `{` and before `}` when destructuring:

`return loaders.hookLastFires.load({ hookGroupId, hookId, filter });`

**Pull request merged**